sdv/evaluation.py

```diff
@@ -52,6 +53,10 @@ def _validate_arguments(synthetic_data, real_data, metadata, root_path, table_na
     elif not isinstance(synthetic_data, type(real_data)):
         raise TypeError('`real_data` and `synthetic_data` must be of the same type')
 
+    # Get table name from metadata for single tables when table_name is not passed
+    if table_name is None and not isinstance(synthetic_data, dict):
+        table_name = list(metadata.to_dict()['tables'].keys())[0]
```

Comment:
I think it would be better to just raise an error and tell them they have to provide the table name in this case